src/utils/viewer/viewer.cpp

```diff
@@ -545,13 +545,44 @@ Viewer::Viewer(const Arguments& arguments)
   objectPickingHelper_ = std::make_unique<ObjectPickingHelper>(viewportSize);
   timeline_.start();
 
-  // Set up per frame profiler
+  /**
+   * Set up per frame profiler to be aware of bottlenecking in processing data
+   * Interpretation: CpuDuration should be less than GpuDuration to avoid GPU
+   * idling, and CpuDuration and GpuDuration should be roughly equal for faster
+   * rendering times
+   *
+   * FrameTime: (Units::Nanoseconds) Time to render per frame, 1/FPS, 2 frame
+   * delay
+   *
+   * CpuDuration: (Units::Nanoseconds) CPU time spent processing events,
+   * physics, traversing SceneGraph, and submitting data to GPU/drivers per
+   * frame
+   * Measured using std::chrono::high_resolution_clock, 1 frame delay
+   *
+   * GpuDuration: (Units::Nanoseconds) GPU time spent rendering data submitted
```

Comment:
measured how much time it takes for the GPU to process all work submitted by the CPU.